### DISCUSSION TITLE
Comparison between Deckset and Marp

### DISCUSSION THREAD

**Discussion by yhatt** [marp-team/marp#68] [General]
> We are making an effort that to be able to have power of expression like a Deckset. 
>
> _Originally posted by @yhatt in https://github.com/yhatt/marp/issues/60#issuecomment-236367300_

Since 2015, the development of Marp has got great influence from [Deckset](https://www.deckset.com/), and there are very similar power of expression in both now.

This is thorough comparison of features between Deckset and Marp. You may find this helpful to migrate into Marp if you have used Deckset!

## Basics

|                     | Deckset                             | Marp                                                                                            |
| ------------------- | ----------------------------------- | ----------------------------------------------------------------------------------------------- |
| Released as         | Paid app (w/ free trial)            | Open-sourced ecosystem                                                                          |
| OS                  | macOS, iOS (Preview)                | Linux, macOS, Windows                                                                           |
| Export to           | PDF, JPG, PNG                       | HTML, PDF, PPTX, JPG, PNG                                                                       |
| Built-in themes     | 19 themes and many color variations | [3 themes and a few color variation](https://github.com/marp-team/marp-core/tree/master/themes) |
| Custom theme        | By tweaking exist themes            | By fully-customizable CSS                                                                       |
| Aspect Ratio Switch | ✅ (GUI)                             | ✅ (`<!-- size: 16:9 -->`)                                                                       |
| Preview Window      | ✅ (GUI)                             | ✅ (`--preview` option in CLI)                                                                   |

## Markdown features

|                                    | Deckset                                          | Marp                                                                                             |
| ---------------------------------- | ------------------------------------------------ | ------------------------------------------------------------------------------------------------ |
| Line breaks                        | ✅ (`<br>` tag to force)                          | ✅ (`<br>` tag to force)                                                                          |
| Tables                             | ✅                                                | ✅                                                                                                |
| Formulas                           | ✅                                                | ✅ (KaTeX / MathJax)                                                                              |
| Syntax highlighting                | ✅                                                | ✅                                                                                                |
| Highlight lines of code            | `[code-highlight: 2]`                            | :bulb: https://github.com/marp-team/marp-vscode/issues/146#issuecomment-640560497                |
| Automatic scaling for code block   | ✅                                                | ✅ (Limited support)                                                                              |
| Emojis                             | `:+1:`                                           | `:+1:`                                                                                           |
| Fit headings                       | `# [fit] xxx`                                    | `# <!--fit--> xxx`                                                                               |
| Footnotes                          | `[^1]: xxx`                                      | :x:                                                                                              |
| Columns                            | `[.column]`                                      | via custom HTML and CSS (#192)                                                                   |
| Presenter notes                    | `^ Presenter notes`                              | `<!-- Presenter notes -->`                                                                       |
| Background image                   | `![](xxx.png)`                                   | `![bg](xxx.png)`                                                                                 |
| Fit background image               | `![fit](xxx.png)`                                | `![bg fit](xxx.png)`                                                                             |
| Multiple background images         | Put multiple `![](xxx.png)`                      | Put multiple `![bg](xxx.png)`                                                                    |
| Background filter                  | ✅ by default (`![original](xxx.png)` to disable) | `![bg opacity](xxx.png)` ([CSS filter](https://marpit.marp.app/image-syntax?id=image-filters))   |
| Zoom in background                 | `![250%](xxx.png)`                               | `![bg 250%](xxx.png)`                                                                            |
| Split slides                       | `![left](xxx.png)`                               | `![bg left](xxx.png)`                                                                            |
| Force apply filter in split slides | `![left filtered](xxx.png)`                      | `![bg left opacity](xxx.png)` (CSS filter)                                                       |
| Global background image            | `background-image: xxx.jpg`                      | `<!-- backgroundImage: url(xxx.jpg) --->`                                                        |
| Inline image                       | `![inline](xxx.png)`                             | `![](xxx.png)`                                                                                   |
| Fill inline image to the slide     | `![inline fill](xxx.png)`                        | `<img src="xxx.png" style="width:100%" />`                                                       |
| Custom scaling for inline image    | `![inline 50%](xxx.png)`                         | `![width:500px](xxx.png)` (Absolute unit only)<br />or `<img src="xxx.png" style="width:50%" />` |
| Image grids                        | Put multiple `![inline](xxx.png)`                | Use custom style                                                                                 |
| Videos                             | `![video.mov]()`                                 | `<video src="video.mov" controls />`                                                             |
| Audios                             | `![audio.mp3]()`                                 | `<audio src="audio.mp3" controls />`                                                             |
| Mermaid.js support                 | ✅ (except ZenUML)                                | :bulb: #207                                                                                      |

## Commands (directives)

|                               | Deckset                                                                                       | Marp                                                                                              |
| ----------------------------- | --------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
| Header                        | :x:                                                                                           | `<!-- header: xxx -->`                                                                            |
| Footer                        | `footer: xxx`                                                                                 | `<!-- footer: xxx -->`                                                                            |
| Slide numbers                 | `slidenumbers: true`                                                                          | `<!-- pagniate: true -->`                                                                         |
| Build lists                   | `build-lists: true`                                                                           | Use `*` / `1)` as bullet list marker ([Fragmented list](https://marpit.marp.app/fragmented-list)) |
| Slide transition              | `slide-transition: true`                                                                      | `<!-- transition: fade -->`                                                                       |
| Use headers as slide dividers | `slide-dividers: #, ##, ###, ####` (global only)                                              | `<!-- headingDivider: 4 -->` or `<!-- headingDivider: [1,2,3,4] -->` (global only)                |
| Specify theme                 | `theme: xxx` (global only)                                                                    | `<!-- theme: xxx -->` (global only)                                                               |
| Customize theme               | [via command](https://docs.decksetapp.com/English.lproj/Customization/02-custom-theming.html) | via CSS (`<style>`)                                                                               |
| Auto scaling                  | `autoscale: true`                                                                             | :x:                                                                                               |
| Hide slides                   | [Use HTML comment](https://docs.decksetapp.com/English.lproj/Presenting/hide-slides.html)     | [Use HTML comment (#162)](https://github.com/orgs/marp-team/discussions/162#discussioncomment-1302928)

**Comment by ctnelson1997** [comment]
Thanks for posting this. For anyone wondering what to do in 2022, 100% go with Marp.

I've inherited a slide deck from a previous instructor that was written with Deckset and it has been an absolute nightmare. Not only is it _paid_ software, but it is only available for Mac, cannot export slides to pptx, and doesn't even have a built-in text editor. I've gone out of my way to buy a Mac Mini, download Deckset, and pay for a license of software that is little more than a glorified visualizer. Deckset is the absolute definition of vendor lock. It is not at all worth the little extra functionality that you get over Marp.

Kudos to the Marp team for their work; I've slowly been salvaging the Decksets into Marp presentations.

**Reply by yhatt** [reply]
I'm happy that you like Marp. We are focusing on vendor-free, framework-free, and frill-free. :smile:
